pkg/cmd/pulumi/import.go

```diff
@@ -212,6 +212,22 @@ func parseImportFile(f importFile, protectResources bool) ([]deploy.Import, impo
 		errs = multierror.Append(errs, fmt.Errorf(format, args...))
 	}
 
+	// Track duplicate resource names and their types.
+	resourceNameMap := map[tokens.QName]tokens.Type{}
+	duplicateIndexes := map[int]struct{}{}
+	duplicateNames := map[tokens.QName]struct{}{}
+	for i, spec := range f.Resources {
+		if existingType, exists := resourceNameMap[spec.Name]; !exists {
+			resourceNameMap[spec.Name] = spec.Type
+		} else if existingType == spec.Type {
```

Comment:
I don't think we should care about type here. We're going to mutate names, let's just fix up all the names regardless of types.